src/http_rest_api_handler.cpp

```diff
@@ -1143,22 +1143,30 @@ Status HttpRestApiHandler::processConfigStatusRequest(std::string& response, Mod
 }
 
 std::string urlDecode(const std::string& encoded) {
-// TODO
-#ifndef _WIN32
-    int output_length;
-    CURL* curl = curl_easy_init();
-    if (curl) {
-        const auto decoded_value = curl_easy_unescape(curl, encoded.c_str(), static_cast<int>(encoded.length()), &output_length);
-        if (decoded_value) {
-            std::string result(decoded_value, output_length);
-            curl_free(decoded_value);
-            curl_easy_cleanup(curl);
-            return result;
+    std::ostringstream decoded;
+    for (size_t i = 0; i < encoded.size(); ++i) {
+        if (encoded[i] == '%') {
+            // Check if the next two characters are valid hex digits
+            if (i + 2 < encoded.size() &&
+                std::isxdigit(static_cast<unsigned char>(encoded[i + 1])) &&
+                std::isxdigit(static_cast<unsigned char>(encoded[i + 2]))) {
```

Comment:
if i + 2 >= encoded.size(), it can be a seqfault
we should add a test with encoded string `test%`

Reply:
if i + 2 is >= encoded.size() we are skip the if and enter else (line 1160-1163)  so there wont be an issue
however, will add test for `test%`

Reply:
added